src/query/sql/src/planner/binder/table_args.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2023 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use common_ast::ast::Expr;
+use common_catalog::table_args::TableArgs;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_expression::type_check::check_literal;
+use common_expression::Scalar;
+
+use crate::plans::ConstantExpr;
+use crate::ScalarBinder;
+use crate::ScalarExpr;
+
+pub async fn bind_table_args(
+    scalar_binder: &mut ScalarBinder<'_>,
+    params: &Vec<Expr>,
+    named_params: &Vec<(String, Expr)>,
+) -> Result<TableArgs> {
+    let mut args = Vec::with_capacity(params.len());
+    for arg in params.iter() {
+        args.push(scalar_binder.bind(arg).await?.0);
+    }
+
+    let mut named_args = Vec::with_capacity(named_params.len());
+    for (name, arg) in named_params.iter() {
+        named_args.push((name.clone(), scalar_binder.bind(arg).await?.0));
+    }
+
+    let positioned_args = args
+        .into_iter()
+        .map(|scalar| match scalar {
+            ScalarExpr::ConstantExpr(ConstantExpr { value, .. }) => Ok(check_literal(&value).0),
+            _ => Err(ErrorCode::Unimplemented(format!(
+                "Unsupported table argument type: {:?}",
+                scalar
+            ))),
+        })
+        .collect::<Result<Vec<_>>>()?;
+
+    let named_args: Vec<(String, Scalar)> = named_args
```

Comment:
Can we just build a `HashMap<String, Scalar>` here?

Reply:
done, thank you!